crates/rover-client/src/operations/graph/check/types.rs

```diff
@@ -1,27 +1,38 @@
 use crate::operations::graph::check::runner::graph_check_mutation;
-use crate::shared::{ChangeSeverity, CheckConfig, GitContext, GraphRef, SchemaChange};
+use crate::shared::{CheckConfig, GitContext, GraphRef};
+
+type MutationInput = graph_check_mutation::CheckSchemaAsyncInput;
+type MutationConfig = graph_check_mutation::HistoricQueryParametersInput;
+type MutationGitContextInput = graph_check_mutation::GitContextInput;
+type MutationVariables = graph_check_mutation::Variables;
+pub(crate) type MutationResponseData = graph_check_mutation::ResponseData;
 
 #[derive(Debug, Clone, PartialEq)]
-pub struct GraphCheckInput {
+pub struct CheckSchemaAsyncInput {
     pub graph_ref: GraphRef,
     pub proposed_schema: String,
     pub git_context: GitContext,
     pub config: CheckConfig,
 }
 
-impl From<GraphCheckInput> for MutationVariables {
-    fn from(input: GraphCheckInput) -> Self {
+impl From<CheckSchemaAsyncInput> for MutationVariables {
+    fn from(input: CheckSchemaAsyncInput) -> Self {
+        let graph_ref = input.graph_ref.clone();
         Self {
             graph_id: input.graph_ref.name,
-            variant: Some(input.graph_ref.variant),
-            proposed_schema: Some(input.proposed_schema),
-            config: input.config.into(),
-            git_context: input.git_context.into(),
+            name: input.graph_ref.variant,
+            input: MutationInput {
+                graph_ref: graph_ref.to_string(),
+                proposed_schema_document: Some(input.proposed_schema),
+                git_context: input.git_context.into(),
+                config: input.config.into(),
+                is_sandbox: false,
+                introspection_endpoint: Some("todo remove this".to_string()), // can't pass none for now because of bug in api
```

Comment:
this is interesting! is there a ticket for tracking nullability for this field? for now we should likely just send an empty string
```suggestion
                // TODO: pass None here when this is nullable (link to issue)
                introspection_endpoint: Some("".to_string()), // can't pass none for now because of bug in api
```

Reply:
bug is fixed so removing this todo!